### PR TITLE
[WB-3390] Identify users in sentry

### PR DESCRIPTION
**Pull request by adrnswanberg**
https://wandb.atlassian.net/browse/WB-3390

### CODE REVIEW

**Comment by adrnswanberg** [comment]
ping

File: wandb/sdk/wandb_run.py [hunk 1]
```diff
@@ -291,7 +291,7 @@ def __init__(self, config=None, settings=None):
 
         # Initial scope setup for sentry. This might get changed when the
         # actual run comes back.
-        sentry_set_scope("user", self._entity, self._project)
+        sentry_set_scope("user", self._entity, self._project, self._settings.email)
```
Comment (raubitsj) [inline wandb/sdk/wandb_run.py:294]: use kwargs when there are this many args please.

File: wandb/sdk/wandb_run.py [hunk 2]
```diff
@@ -613,7 +613,13 @@ def _set_run_obj(self, run_obj):
             self.summary.update(summary_dict)
         self.history._update_step()
         # TODO: It feels weird to call this twice..
-        sentry_set_scope("user", run_obj.entity, run_obj.project, self._get_run_url())
+        sentry_set_scope(
+            "user",
```
Comment (raubitsj) [inline wandb/sdk/wandb_run.py:617]: kwargs

**Review by raubitsj — APPROVED** [review]
small nit

File: wandb/sdk/wandb_run.py (hunk 1 above)
Comment (adrnswanberg) [reply wandb/sdk/wandb_run.py:294]: You mean like pack them into a dictionary and use the `**` operator?

Comment (raubitsj) [reply wandb/sdk/wandb_run.py:294]: Sorry, more precisely, keyword arguments

I was thinking at least:
```
sentry_set_scope("user", self._entity, self._project, email=self._settings.email)
```
but you might want to use entity=self._entity, project=self._project also.

I think as soon as you get over 2 or 3 arguments, i think positional arguments are too error prone.   And definitely if you have more than one optional argument, positional arguments should never be used.

**Pull request merged**